src/interpreter_funcs/voxel_metaballs.rs

```diff
@@ -0,0 +1,303 @@
+            \n\
+            The input meshes will be marked used and thus invisible in the viewport. \
+            They can still be used in subsequent operations.\n\
+            \n\
+            The resulting mesh geometry will be named 'Metaball Mesh'.",
+            return_value_name: "Metaball Mesh",
+        }
+    }
+
+    fn flags(&self) -> FuncFlags {
+        FuncFlags::PURE
+    }
+
+    fn param_info(&self) -> &[ParamInfo] {
+        &[
+            ParamInfo {
+                name: "Mesh 1",
+                description: "First input mesh.",
+                refinement: ParamRefinement::Mesh,
+                optional: false,
+            },
+            ParamInfo {
+                name: "Mesh 2",
+                description: "Second input mesh.",
+                refinement: ParamRefinement::Mesh,
+                optional: false,
+            },
+            ParamInfo {
+                name: "Voxel Size",
+                description: "Size of a single cell in the regular three-dimensional voxel grid.\n\
+                \n\
+                High values produce coarser results, low values may increase precision but produce \
+                heavier geometry that significantly affect performance. Too high values produce \
+                single large voxel, too low values may generate holes in the resulting geometry.",
+                refinement: ParamRefinement::Float3(Float3ParamRefinement {
+                    min_value: Some(0.005),
+                    max_value: None,
+                    default_value_x: Some(0.1),
+                    default_value_y: Some(0.1),
+                    default_value_z: Some(0.1),
+                }),
+                optional: false,
+            },
+            ParamInfo {
+                name: "Fill Closed Volumes",
+                description: "Treats the insides of watertight mesh geometries as volumes.\n\
+                \n\
+                If this option is off, the resulting voxelized mesh geometries will have two \
+                separate mesh shells: one for outer surface, the other for inner surface of \
+                hollow watertight mesh.",
+                refinement: ParamRefinement::Boolean(BooleanParamRefinement {
+                    default_value: true,
+                }),
+                optional: false,
+            },
+            ParamInfo {
+                name: "Distance Multiplier",
+                description: "Defines the speed of volume falloff.\n\
+                \n\
+                Lowe values mean smoother and larger results.",
+                refinement: ParamRefinement::Float(FloatParamRefinement {
+                    default_value: Some(0.5),
+                    min_value: Some(0.0001),
+                    max_value: Some(1.0),
+                }),
+                optional: false,
+            },
+            ParamInfo {
+                name: "Volume range",
+                description: "Max value 2 = fill volumes",
```

Comment:
Are you sure about this comment?

Reply:
this is incredibly tricky, not only regarding the comment. the values can be unexpected and actually i don't know how to set them properly. the issue roots in the falloff functions, for voxels inside volumes. actually, i don't know what should be the correct mathematical solution.
more in person